src/components/TextInputWithLabel.js

```diff
@@ -0,0 +1,39 @@
+import _ from 'underscore';
+import React from 'react';
+import {Text, TextInput} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+
+const propTypes = {
+    // Label text
+    label: PropTypes.string.isRequired,
+
+    // Whether this field has an error
+    hasError: PropTypes.bool,
+
+    // Text to show if there is an error
+    errorText: PropTypes.string,
+};
+
+const defaultProps = {
+    hasError: false,
+    errorText: '',
+};
+
+const TextInputWithLabel = props => (
+    <>
+        <Text style={[styles.formLabel]}>{props.label}</Text>
+        <TextInput
+            style={[styles.textInput, styles.mb1]}
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {..._.omit(props, ['label'])}
```

Comment:
Should this also omit `hasError` and `errorText`? I think so.

Reply:
In my review I suggested just passing the props in and not worrying about the omission. AFAIK passing `label`, `hasError`, `errorText` to a regular `TextInput` should have no effect on the `TextInput`.